### PR TITLE
refactor(api): save and load pipette offset data

### LINKED ISSUE #6416: Offset Calibration: Load and store offset data
Add functions to the api to load and store pipette offset data based on pipette serial.

- The offsets on disk should be named by pipette serial and by mount
- The offsets on disk should include a timestamp of when calibration was executed and the calibration hash and URI of the tiprack used to execute calibration as well as the 3d offset
- The function that loads the data by serial and mount should also return the metadata (and should be public - this will be used to expose the data to the user)

### PR DESCRIPTION
**Pull request by ahiuchingau**
closes #6416

# Overview
This PR adds functions on the api side to load and save pipette calibration data.

# Changelog
- add functions in `/calibration_storage` to save and load pipette offset data
- add a load function in `hardware_controller/robot_calibration` to allow more straightforward access to offset data

# Review requests
Data format looks ok?
Two follow-up PRs will take care of (1) adding these functions in the pipette calibration flow and (2) using the new pipette offset data in hardware control motion, once this PR is merged.

# Risk assessment
Low, new code is not hooked up yet

### CODE REVIEW

File: api/src/opentrons/calibration_storage/modify.py [hunk 1]
```diff
@@ -192,6 +193,7 @@ def save_robot_deck_attitude(
         pip_id: typing.Optional[str],
         lw_hash: typing.Optional[str]):
     robot_dir = config.get_opentrons_path('robot_calibration_dir')
+    print(robot_dir)
```
Comment (sfoster1) [inline api/src/opentrons/calibration_storage/modify.py:196]: debug print, should remove

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Need to remove a debug print but looks good other than that

File: api/src/opentrons/calibration_storage/modify.py (hunk 1 above)
Comment (ahiuchingau) [reply api/src/opentrons/calibration_storage/modify.py:196]: oops

**Review by sfoster1 — APPROVED** [review]
Looks good to me!

**Pull request merged**